systems/robotInterfaces/QPLocomotionPlan.m

```diff
@@ -44,47 +44,42 @@
       obj.robot = robot;
       S = load(obj.robot.fixed_point_file);
       obj.qtraj = S.xstar(1:obj.robot.getNumPositions());
-      obj.default_qp_input = atlasControllers.QPInputConstantHeight();
-      obj.default_qp_input.whole_body_data.q_des = zeros(obj.robot.getNumPositions(), 1);
+      obj.default_qp_input_ = atlasControllers.QPInputConstantHeight();
```

Comment:
that also sounds good to me. 

On May 8, 2015, at 1:46 PM, Twan Koolen <notifications@github.com> wrote:

> In systems/robotInterfaces/QPLocomotionPlan.m:
> 
> > @@ -44,47 +44,42 @@
> >        obj.robot = robot;
> >        S = load(obj.robot.fixed_point_file);
> >        obj.qtraj = S.xstar(1:obj.robot.getNumPositions());
> > -      obj.default_qp_input = atlasControllers.QPInputConstantHeight();
> > -      obj.default_qp_input.whole_body_data.q_des = zeros(obj.robot.getNumPositions(), 1);
> > +      obj.default_qp_input_ = atlasControllers.QPInputConstantHeight();
> While establishing the interface for the C++ QPLocomotionPlan I needed to make sure that the fields in the Matlab version were not being modified externally. To do that I made the fields private and added getters where necessary. The idea was to have the C++ QPControllerPlan implement these getters as well (while having the underlying data live in the C++ object).
> 
> In the end, it turned out that only duration and start_time needed to be publicly accessible for any QPControllerPlan, so those are the methods I added to the QPControllerPlan.m file. default_qp_input is a method in QPControllerPlanMatlabImplementation, which inherits from QPControllerPlan.
> 
> Renaming the fields and having the getter functions have the same names that the fields had previously was a way to avoid making changes in many files in both drake and DRC ata precarious point. Now that everything's working, should we rename the functions in this pull request? i.e.:
> 
> duration --> getDuration in QPControllerPlan.m
> start_time --> getStartTime in QPControllerPlan.m
> default_qp_input --> getDefaultQPInput in QPControllerPlanMatlabImplementation
> gain_set --> getGainSet in QPControllerPlanMatlabImplementation
> —
> Reply to this email directly or view it on GitHub.
>